vanila/src/components/Table/Pagination.ts

```diff
@@ -0,0 +1,160 @@
+import { FilmDTO } from '../../DTO/filmDTO';
+
+import { CollectionDocument } from '../../Interfaces/collectionDocument';
+
+import { setFilmsInTable } from './Table';
+
+const paginationElement = document.createElement('ul');
+paginationElement.classList.add('pagination');
+paginationElement.innerHTML = `
+<li class="disabled btn_prev">
+  <a>
+    <i class="material-icons">chevron_left</i>
+  </a>
+</li>
+<section class="pages"></section>
+<li class="waves-effect btn_next">
+  <a>
+    <i class="material-icons">chevron_right</i>
+  </a>
+</li>
+`;
+let currentPage = 1;
+const records_per_page = 3;
+
+let filmsDocs: CollectionDocument<FilmDTO>[] = [];
+
+/**
+ * Change page.
+ * @param pageNum Number page.
+ */
+function pageHandler(pageNum: number): void {
+  if (currentPage !== pageNum) {
+    setFilmsInTable(changePage(pageNum));
+  }
+}
+
+/**
+ * Go to prev page.
+ */
+function goPrevPage(): CollectionDocument<FilmDTO>[] {
+  if (currentPage > 1) {
+    return changePage(currentPage - 1);
+  }
+  return [];
+
+}
+
+/**
+ * Go to next page.
+ */
+function goNextPage(): CollectionDocument<FilmDTO>[] {
+  if (currentPage < getNumPages()) {
+    return changePage(currentPage + 1);
+  }
+  return [];
+
+}
+
+/**
+ * Change page number.
+ * @param page Num of page.
+ */
+export function changePage(page: number): CollectionDocument<FilmDTO>[] {
+  if (getNumPages() !== 0) {
+    const buttonNext = <HTMLLIElement>paginationElement.querySelector('.btn_next');
+    const buttonPrev = <HTMLLIElement>paginationElement.querySelector('.btn_prev');
+    const data: CollectionDocument<FilmDTO>[] = [];
+    let selectedPage: number;
+
+    // Validate page
+    if (page < 1) {
+      selectedPage = 1;
+    } else if (page > getNumPages()) {
+      selectedPage = getNumPages();
+    } else {
+      selectedPage = page;
+    }
+
+    for (let i = (selectedPage - 1) * records_per_page; i < (selectedPage * records_per_page); i++) {
```

Comment:
I think you can take the calculations out of the loop. Imagine if you had a resource-intensive calculation in a function that would be called on each iteration of the loop.

```suggestion
    const offset = (selectedPage * records_per_page);
    for (let i = (selectedPage - 1) * records_per_page; i < offset; i++) {
```